test/test_dynamic_shapes.py

```diff
@@ -944,6 +942,61 @@ def assert_not_optimized(sym):
         assert_not_optimized(b)
         assert_not_optimized(a + b)
 
+    def test_sympy_optimized_add_with_co(self):
+        shape_env = ShapeEnv()
+        s0 = create_symint(shape_env, 2)
+        s1 = create_symint(shape_env, 3)
+        s2 = create_symint(shape_env, 4)
+        sum = 4 * s0 + 4 * s1
+
+        def assert_optimized(sym):
+            self.assertEqual(sym.node._optimized_summation_co, 4)
+
+        def assert_not_optimized(sym):
+            self.assertEqual(getattr(sym.node, "_optimized_summation_co", None), None)
```

Comment:
I'm not a big fan of tests that are reaching internal implementation details like this, because they invariably need to be rewritten when you change the internal implementation. Instead, consider defining some sort of invariant checker on SymNode and just calling that.

Reply:
make sense